mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -127,14 +135,27 @@ class StringIndexerModel private[ml] (
       if (labelToIndex.contains(label)) {
         labelToIndex(label)
       } else {
-        // TODO: handle unseen labels
-        throw new SparkException(s"Unseen label: $label.")
+        if (!getSkipInvalid) {
+          throw new SparkException(s"Unseen label: $label.")
+        } else {
```

Comment:
I don't think the two different exceptions add any new information (the user already knows if the model is configured to skip invalid or not). Perhaps we can collapse this into a single exception?